Automattic/wp-calypso
# calypso-build: add webpack config for output.jsonpFunction

#### Changes proposed in this Pull Request

* Add configuration for output.webpackjsonpFunction in the calypso-build package

The [webpack docs state](https://v4.webpack.js.org/configuration/output/#outputjsonpfunction)

> If multiple webpack runtimes (from different compilations) are used on the same webpage, there is a risk of conflicts of on-demand chunks in the global namespace.

This seems to be a source of WordPress plugin conflicts, when multiple plugins use webpack to build js. For example: https://github.com/Automattic/jetpack/issues/17289

Adding this config allows plugins that use calypso-build's webpack config, like Jetpack, to customize the webpack jsonp global variable name and avoid these conflicts.

Note that this config is changing to `chunkLoadingGlobal` in [webpack 5](https://webpack.js.org/configuration/output/#outputchunkloadingglobal), so we might want to address that in some way, as well.

#### Testing instructions

* Add `output-jsonp-function` to a webpack build process that uses `getWebpackConfig` from calypso-build (Jetpack example: https://github.com/Automattic/jetpack/pull/17312/commits/33815b44dd0d69f039aa543ef3dc71f37e338820)
* Check the output js, and see that `window["webpackJsonp"]` uses what you specified for the global variable

Required for fixing https://github.com/Automattic/jetpack/issues/17289
